src/main/QueryServer.cpp

```diff
@@ -132,7 +133,15 @@ QueryServer::getLedgerEntryRaw(std::string const& params,
 
     if (!keys.empty())
     {
-        auto& bl = *mBucketListSnapshots.at(std::this_thread::get_id());
+        auto& snapshotPtr = mBucketListSnapshots.at(std::this_thread::get_id());
+        if (snapshotPtr->getLedgerSeq() <
```

Comment:
This block is non-atomic, which is probably not a big deal here, but could be problematic if this pattern reused in the place where this does matter. It would be safer to provide a single function `maybeUpdateSnapshot` that atomically verifies the ledger sequence and assigns the new snapshot if necessary. At the very least that would be more efficient as only 1 lock would be necessary.